### PR TITLE
Add Edit Goal Feature, Fix Parent-Child Goal Relationship, and Add Goal Tests

### PR DESCRIPTION
**Pull request by ConnorBoulais**
This PR adds a feature that allows the user to edit goals. It also allows the user to check off parent goals that have no sub goals, and automatically marks parent goals as complete when all sub goals are completed.

I also added some styling and tests for the Goals components.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -84,7 +85,7 @@ function App() {
   /* Main ApolloClient object, any apollo configuration
    * for our frontend will likely live here. */
   const client = new ApolloClient({
-    uri: process.env.REACT_APP_PROD_URI,
```
Comment (samiur98) [inline src/App.js:87]: A very necessary change.

File: src/Components/FillBlank.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ function FillBlank() {
      {myname: 'B1'}];
 
    const textMap = (data) => {
-    return data.map(({name}) => (<input type="text" id="cname" name="cname" value={name} onChange={event => setName(event.target.value)}/>));
+    return data.map(({name}, index) => (<input key={index} type="text" id="cname" name="cname" value={name} onChange={event => setName(event.target.value)}/>));
```
Comment (samiur98) [inline src/Components/FillBlank.js:18]: Typo with cname?

File: src/Components/Goals/Goal.js [hunk 3]
```diff
@@ -24,6 +33,19 @@ function Goal(props){
...
   const [open, setOpen] = useState(localStorage.getItem(goal_open_key) === "open");
 
+
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
```
Comment (samiur98) [inline src/Components/Goals/Goal.js:43]: Looks good.

File: src/Components/Goals/GoalForm.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                KeyboardButtonProps={{
+                  'aria-label': 'change date',
+                }}
+              />
+            </div>
+          </div>
+          <br/>
+
+          <h2 style={{display: "flex"}}>Add Sub Goals</h2>
+
```
Comment (samiur98) [inline src/Components/Goals/GoalForm.js:123]: Good Job!

File: src/Tests/Goals.test.js [hunk 5]
```diff
@@ -0,0 +1,240 @@
+import { render, fireEvent, screen, cleanup, act } from '@testing-library/react';
+import { MockedProvider } from '@apollo/client/testing';
+import GoalList from '../Components/Goals/GoalList';
+import { GET_ALL_GOALS, EDIT_OR_CREATE_GOAL } from '../gqlQueries';
+import { InMemoryCache } from '@apollo/client';
+
+const GOAL1 = {
+	id: '001',
```
Comment (samiur98) [inline src/Tests/Goals.test.js:8]: Solid Tests.

**Review by samiur98 — APPROVED** [review]
Good job!

File: src/Components/Goals/Goal.js [hunk 6]
```diff
@@ -1,8 +1,17 @@
...
-import { EDIT_OR_CREATE_GOAL } from '../gqlQueries';
+import { EDIT_OR_CREATE_GOAL } from '../../gqlQueries';
 import { gql } from '@apollo/client';
+import IconButton from '@material-ui/core/IconButton';
+import EditIcon from '@material-ui/icons/Edit';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import { makeStyles } from '@material-ui/core/styles';
+
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
```
Comment (tddumont) [inline src/Components/Goals/Goal.js:12]: These icons look great!

File: src/Components/Goals/GoalsScreen.css [hunk 7]
```diff
@@ -71,19 +64,8 @@
...
-  grid-template-columns:2fr 2fr 1fr;
-  background-color: #889cb9;
-  border: 1px;
-  grid-column:2/3;
-  align-items: center;
-  padding: 0;
-  margin: 0;
-}
-
-.subGoal:nth-child(odd)  {
```
Comment (tddumont) [inline src/Components/Goals/GoalsScreen.css:83]: I agree that the styling looks better without the alternating colors.

File: src/Tests/Goals.test.js [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+	}
+];
+
+afterEach(() => {
+	cleanup();
+});
+
+test("test for loading message", () => {
+	const { getByText } = render(
+		<MockedProvider mocks={mocks}>
```
Comment (tddumont) [inline src/Tests/Goals.test.js:126]: Great job figuring out how to mock queries!

**Review by tddumont — APPROVED** [review]
As usual, great work! I like the new look of the Goals!

**Pull request merged**